geoapps/base/dash_application.py

```diff
@@ -37,21 +45,101 @@ def __init__(self, **kwargs):
             external_stylesheets=external_stylesheets,
         )
 
+    def update_object_options(
+        self, filename: str, contents: str, trigger: str = None
+    ) -> (dict, list):
+        """
+        This function is called when a file is uploaded. It sets the new workspace, sets the dcc ui_json component,
+        and sets the new object options.
+
+        :param filename: Uploaded filename. Workspace or ui.json.
+        :param contents: Uploaded file contents. Workspace or ui.json.
+        :param trigger: Dash component which triggered the callback.
+
+        :return ui_json: Uploaded ui_json.
+        :return options: New dropdown options.
+        """
+        ui_json, options = no_update, no_update
+
+        if trigger is None:
+            trigger = callback_context.triggered[0]["prop_id"].split(".")[0]
+        if contents is not None or trigger == "":
+            if filename is not None and filename.endswith(".ui.json"):
+                content_type, content_string = contents.split(",")
+                decoded = base64.b64decode(content_string)
+                ui_json = json.loads(decoded)
+                self.workspace = Workspace(ui_json["geoh5"])
+                ui_json = self.load_ui_json(ui_json)
+            elif filename is not None and filename.endswith(".geoh5"):
+                content_type, content_string = contents.split(",")
+                decoded = io.BytesIO(base64.b64decode(content_string))
+                self.workspace = Workspace(decoded)
+                ui_json = no_update
+            elif trigger == "":
+                ifile = InputFile(
+                    ui_json=self.params.input_file.ui_json,
+                    validation_options={"disabled": True},
+                )
+                ifile.update_ui_values(self.params.to_dict())
+                ui_json = self.load_ui_json(ifile.ui_json)
+            options = [
+                {"label": obj.parent.name + "/" + obj.name, "value": obj.name}
```

Comment:
Can we work with uuid for value? It would be safer against duplicate entity names.